src/Provider/Nominatim/Nominatim.php

```diff
@@ -169,7 +174,23 @@ private function xmlResultToArray(\DOMElement $resultNode, \DOMElement $addressN
             $builder->setBounds($bounds['south'], $bounds['west'], $bounds['north'], $bounds['east']);
         }
 
-        return $builder->build();
+        if (false === $reverse) {
```

Comment:
Do we really need this separation? It seams like most the lines are the same. 
I suggest to remove the `$reverse` parameter.

Reply:
Yes, I don't really like that solution with `reverse` parameter either but the API answer from geocoding or reverse geocoding is quite different so the extraction of the files is also different :

**Geocoding:**
```$location = $location->withDisplayName($resultNode->getAttribute('display_name'));```
**Reverse geocoding:**
```$location = $location->withDisplayName($resultNode->nodeValue);```

Any suggestion to improve/simplify that ?

Reply:
I see.. Lets improve it by only putting the differences in the if-else-statement. It is a private function so we may improve it later if we come up with anything.

Reply:
👍 I'll take care of that today !